src/openforms/forms/tests/variables/test_model.py

```diff
@@ -269,3 +293,303 @@ def test_user_defined_variables_cast_initial_value(self):
                     self.assertEqual(
                         expected_values[data_type][index], variable.initial_value
                     )
+
+
+class FormVariableManagerTests(TestCase):
+
+    @tag("gh-5084", "slow")
+    def test_performance_upsert_single_form_definition(self):
+        many_components = [
+            {
+                "type": "textfield",
+                "key": f"component-{i}",
+                "label": f"Field {i}",
+                "id": f"comp{i}",
+                "prefill": {
+                    "plugin": "demo",
+                    "attribute": "random",
+                    "identifierRole": IdentifierRoles.main,
+                },
+            }
+            for i in range(1000)
+        ]
+        fd = FormDefinition.objects.create(
+            slug="slak",
+            configuration={"components": many_components},
+            is_reusable=False,
+        )
+        form = FormFactory.create()
+        FormStep.objects.create(form=form, form_definition=fd, order=0, slug="slak")
+        assert not FormVariable.objects.exists()
+
+        with (
+            self.subTest("initial inserts"),
+            # 5 seconds is giving *a lot* of credit, but before the performance patch
+            # this was taking 10+ seconds. In isolation, this takes 100-400ms.
+            assert_execution_time(self, seconds=5),
```

Comment:
I have not used this kind of tests and I guess sometime will maybe fail right?I mean concerning the time we specify. And maybe this is something deliberate since we can understand that there is a new problem?

Reply:
yes, it's a risk for flakiness, so I'm making sure I'm using different orders of magnitude. i.e. usually this completes in about 0.5s, so I'm taking an upper limit of 10 times that to account for differences in hardware/CPU speed etc.

The performance was so bad that before the patch 10+s was spent, which fit that (terrible) order of magnitude.

I'm also investigating codspeed.io to not have to rely on these kind of flaky constructs, but instead have a dedicated tool in our CI pipeline to catch performance regressions.